cmd/ftl/cmd_serve.go

```diff
@@ -105,6 +118,66 @@ func (s *serveCmd) Run(ctx context.Context) error {
 	return nil
 }
 
+func setupDB(ctx context.Context) (*string, error) {
+	logger := log.FromContext(ctx)
+
+	dsn := fmt.Sprintf("postgres://postgres:secret@localhost:5433/%s?sslmode=disable", ftlDBName)
+
+	logger.Infof("Checking for FTL database")
+
+	nameFlag := fmt.Sprintf("name=^/%s$", ftlDBName)
+	output, err := exec.Capture(ctx, ".", "docker", "ps", "-a", "--filter", nameFlag, "--format", "{{.Names}}")
+	if err != nil {
+		logger.Errorf(err, "%s", output)
+		return nil, errors.WithStack(err)
+	}
+
+	recreate := false
+
+	if len(output) == 0 {
+		logger.Infof("Creating FTL database")
+
+		err = exec.Command(ctx, logger.GetLevel(), "./", "docker", "run",
+			"-d", // run detached so we can follow with other commands
+			"--name", ftlDBName,
+			"--user", "postgres",
+			"--restart", "always",
+			"-e", "POSTGRES_PASSWORD=secret",
+			"-p", "5433:5432",
```

Comment:
Re. the above comment, changing this to `-p 5432` will dynamically allocate a port. I think you could then pull the real port out of the `docker inspect` below.